plugins/MosaicCamera/src/MosaicCamera.cpp

```diff
@@ -0,0 +1,560 @@
+/*
+ * Copyright (C) 2024 Josh Meyers
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA  02110-1335, USA.
+ */
+
+#include "StelProjector.hpp"
+#include "StelApp.hpp"
+#include "StelCore.hpp"
+#include "StelFileMgr.hpp"
+#include "StelGui.hpp"
+#include "StelGuiItems.hpp"
+#include "StelLocaleMgr.hpp"
+#include "StelModuleMgr.hpp"
+#include "StelMovementMgr.hpp"
+#include "MosaicCamera.hpp"
+#include "MosaicCameraDialog.hpp"
+#include "StelUtils.hpp"
+#include "StelPainter.hpp"
+#include "CustomObject.hpp"
+
+#include <QDebug>
+#include <QDir>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QJsonObject>
+#include <QResource>
+#include <QByteArray>
+
+/*************************************************************************
+ This method is the one called automatically by the StelModuleMgr just
+ after loading the dynamic library
+*************************************************************************/
+StelModule* MosaicCameraStelPluginInterface::getStelModule() const
+{
+	return new MosaicCamera();
+}
+
+StelPluginInfo MosaicCameraStelPluginInterface::getPluginInfo() const
+{
+	StelPluginInfo info;
+	info.id = "MosaicCamera";
+	info.displayedName = "Mosaic Camera";
+	info.authors = "Josh Meyers";
+	info.contact = "jmeyers314@gmail.com";
+	info.description = "Mosaic Camera Overlay";
+	info.version = MOSAICCAMERA_PLUGIN_VERSION;
+	info.license = MOSAICCAMERA_PLUGIN_LICENSE;
+	return info;
+}
+
+/*************************************************************************
+ Constructor
+*************************************************************************/
+MosaicCamera::MosaicCamera()
+    : toolbarButton(Q_NULLPTR)
+{
+    setObjectName("MosaicCamera");
+    configDialog = new MosaicCameraDialog();
+    StelApp &app = StelApp::getInstance();
+    conf = app.getSettings();
+    gui = dynamic_cast<StelGui*>(app.getGui());
+}
+
+/*************************************************************************
+ Destructor
+*************************************************************************/
+MosaicCamera::~MosaicCamera()
+{
+    delete configDialog;
+}
+
+void MosaicCamera::loadSettings()
+{
+    conf->beginGroup("MosaicCamera");
+    currentCamera = conf->value("currentCamera").toString();
+    setFlagShowButton(conf->value("showButton").toBool());
```

Comment:
You can safely add a default-true here:

`setFlagShowButton(conf->value("showButton", true).toBool());`